<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CVariableGeneralDialog</class>
 <widget class="QDialog" name="CVariableGeneralDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>543</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>9</x>
     <y>10</y>
     <width>381</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>名称</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="nameEdit">
   <property name="geometry">
    <rect>
     <x>9</x>
     <y>27</y>
     <width>381</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>9</x>
     <y>52</y>
     <width>381</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>变量类型</string>
   </property>
  </widget>
  <widget class="QComboBox" name="variableTypeBox">
   <property name="geometry">
    <rect>
     <x>9</x>
     <y>80</y>
     <width>381</width>
     <height>25</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>虚拟变量</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>I/O变量</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>中间变量</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>9</x>
     <y>102</y>
     <width>381</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>工程数据类型</string>
   </property>
  </widget>
  <widget class="QComboBox" name="dataTypeBox">
   <property name="geometry">
    <rect>
     <x>9</x>
     <y>130</y>
     <width>381</width>
     <height>25</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>Boolean</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Char</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Byte</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Int16</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>UInt16</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Int32</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>UInt32</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Int64</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>UInt64</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Float</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Double</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Datetime</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>String</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label_8">
   <property name="geometry">
    <rect>
     <x>9</x>
     <y>155</y>
     <width>381</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>小数位精度（位）</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="precisionEdit">
   <property name="geometry">
    <rect>
     <x>9</x>
     <y>180</y>
     <width>381</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>9</x>
     <y>202</y>
     <width>381</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>设备</string>
   </property>
  </widget>
  <widget class="QComboBox" name="deviceBox">
   <property name="geometry">
    <rect>
     <x>9</x>
     <y>230</y>
     <width>381</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>9</x>
     <y>250</y>
     <width>381</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>地址/表达式</string>
   </property>
  </widget>
  <widget class="QPushButton" name="addressButton">
   <property name="geometry">
    <rect>
     <x>320</x>
     <y>280</y>
     <width>71</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>...</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="addressEdit">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>280</y>
     <width>301</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>300</y>
     <width>381</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>原始数据类型</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>410</y>
     <width>381</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>描述</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="descriptionEdit">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>440</y>
     <width>382</width>
     <height>31</height>
    </rect>
   </property>
   <property name="placeholderText">
    <string/>
   </property>
  </widget>
  <widget class="QComboBox" name="originalDataTypeBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>330</y>
     <width>381</width>
     <height>25</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>Boolean</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Char</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Byte</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Int16</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>UInt16</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Int32</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>UInt32</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Int64</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>UInt64</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Float</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Double</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Datetime</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>String</string>
    </property>
   </item>
  </widget>
  <widget class="QTableWidget" name="addressTableWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>381</width>
     <height>271</height>
    </rect>
   </property>
   <property name="textElideMode">
    <enum>Qt::ElideRight</enum>
   </property>
   <property name="columnCount">
    <number>2</number>
   </property>
   <column/>
   <column/>
  </widget>
  <widget class="QLabel" name="label_9">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>350</y>
     <width>381</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>单位</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="unitEdit">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>380</y>
     <width>382</width>
     <height>31</height>
    </rect>
   </property>
   <property name="placeholderText">
    <string/>
   </property>
  </widget>
  <widget class="QTextEdit" name="extEdit">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>500</y>
     <width>382</width>
     <height>31</height>
    </rect>
   </property>
   <property name="placeholderText">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_10">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>470</y>
     <width>381</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>扩展</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
